win/WIN_UIelement.h

```diff
@@ -35,19 +26,25 @@ namespace win
 		gfx::Colour getForegroundColour() const { return foregroundColour_; }
 		void getForegroundColour(uint8_t rgba[]);
 		void getBackgroundColour(uint8_t rgba[]);
-		void swapColours();
+		virtual void swapColours();
 
 		void setRect(const gfx::Rectangle & rect);
 		gfx::Rectangle getRect() const { return rect_; }
 		const char* getName() const { return name_.c_str(); }
-		UIelement* getParent() { return parent_; }
+		UIelement* getParent() const { return parent_; }
 		void setParent(UIelement * parent);
```

Comment:
We need an interface

void setEnabled(bool enable);
bool isEnabled() const;

to disable and enable the control when so required. The internal logic and rendering should reflect this (eg gray out the control when disabled and prevent interaction).

I have added a story to the board to this end for the next sprint.
**MUST**